test/src/unittests/tonal/test_tonicindianartmusic.py

```diff
@@ -33,66 +34,88 @@ def testInvalidParam(self):
         self.assertConfigureFails(TonicIndianArtMusic(), { 'maxTonicFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'minTonicFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': 0 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberHarmonics': -1 })        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'numberSaliencePeaks': 0})
+        self.assertConfigureFails(TonicIndianArtMusic(), { 'numberSaliencePeaks': 16})        
         self.assertConfigureFails(TonicIndianArtMusic(), { 'referenceFrequency': -1 })
         self.assertConfigureFails(TonicIndianArtMusic(), { 'sampleRate':   -1 })
 
-    def testZeros(self):
-        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()(zeros(1024)))   
+    def testEmpty(self):
+        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()([]))   
 
-    def testOnes(self):
-        referenceTonic =108.86099243164062     
-        tonic   = TonicIndianArtMusic()(ones(1024))
-        self.assertAlmostEqual(referenceTonic, tonic, 8)
+    def testSilence(self):
+        silence = np.zeros(int(np.abs(np.random.randn()) * 30. * 44100))
+        self.assertRaises(RuntimeError, lambda: TonicIndianArtMusic()(silence))
 
-    def testNegativeInput(self):
-        tonic = TonicIndianArtMusic()([-1]*1024)
-        referenceTonic =108.86099243164062     
-        tonic   = TonicIndianArtMusic()([-1]*1024)
-        self.assertAlmostEqual(referenceTonic, tonic, 8)
+    def testOnes(self):
+        # FIXME: The tonic value 117.36 is not returned everytime. Sometimes 55 is returned.
+        # Also, different values can be obtained when tweaking minTonic, maxTonic and reference frequencies.
+        referenceTonic = 117.36 
+        tonic   = TonicIndianArtMusic()(ones(4096))
+        self.assertAlmostEqual(tonic, referenceTonic, 5)
 
     def testRegression(self):
         audio = MonoLoader(filename = join(testdata.audio_dir, 'recorded/vignesh.wav'),
                             sampleRate = 44100)()
         referenceTonic = 102.74                                       
         tonic = TonicIndianArtMusic()(audio)
-        self.assertAlmostEqual(referenceTonic, tonic, 6)
+        self.assertAlmostEqual( tonic, referenceTonic, 6)
+        start_zero = np.zeros(int(np.abs(np.random.randn()) * 30. * 44100))
+        end_zero = np.zeros(int(np.abs(np.random.randn()) * 30. * 44100))
+        # Check result is the same with appended silences
+        real_audio = np.hstack([start_zero, audio, end_zero])
+        tonic = TonicIndianArtMusic()(real_audio)
+        # FIXME: tonic is sometimes 102.74112701416016 and other times  154.9361114501953
+        self.assertAlmostEqual(tonic, referenceTonic, 5)
+
+    def testWhiteNoise(self):
+        from numpy.random import uniform
+        sig = array(uniform(size=10000))
+        tonic  = TonicIndianArtMusic()(sig)
```

Comment:
TonicIndianArtMusic expects an input audio signal, so this test is invalid (the uniform distribution is for the frequency domain, not the time domain).

On the other side, not sure it is really necessary either.

Reply:
removed